src/types/block.cairo

```diff
@@ -43,18 +43,25 @@ pub struct Header {
 pub impl BlockHashImpl of BlockHash {
     /// Compute hash of the block header given the missing fields.
     fn hash(self: @Header, prev_block_hash: Hash, merkle_root: Hash) -> Hash {
-        let mut header_data_bytes: ByteArray = Default::default();
-
-        header_data_bytes.append_word_rev((*self.version).into(), 4);
-        let best_block_hash: ByteArray = prev_block_hash.into();
-        header_data_bytes.append(@best_block_hash);
-        let merkle_root: ByteArray = merkle_root.into();
-        header_data_bytes.append(@merkle_root);
-        header_data_bytes.append_word_rev((*self.time).into(), 4);
-        header_data_bytes.append_word_rev((*self.bits).into(), 4);
-        header_data_bytes.append_word_rev((*self.nonce).into(), 4);
-
-        double_sha256_byte_array(@header_data_bytes)
+        let mut header_data_u32: Array<u32> = array![];
+
+        header_data_u32.append(self.u32_byte_reverse(*self.version));
+        header_data_u32.append_span(prev_block_hash.value.span());
+        header_data_u32.append_span(merkle_root.value.span());
+
+        header_data_u32.append(self.u32_byte_reverse(*self.time));
+        header_data_u32.append(self.u32_byte_reverse(*self.bits));
+        header_data_u32.append(self.u32_byte_reverse(*self.nonce));
+
+        double_sha256_u32_array(header_data_u32)
+    }
+
+    fn u32_byte_reverse(self: @Header, word: u32) -> u32 {
```

Comment:
Please put it to a new module `utils::numeric` and add a unit test there
Btw, have you checked if this approach is more efficient in terms of steps https://github.com/starkware-libs/cairo/issues/3589#issuecomment-1629042387 ?

Reply:
Got it. I'll do it in some hours. I'm gonna be busy this morning.

Reply:
Hello again mate @m-kus. Honestly I did not check if it ss more efficient in terms of steps.

And when u say docstring, you mean something like this?
![image](https://github.com/user-attachments/assets/0959c5e5-6853-4e42-b1e7-462905eb336c)

Reply:
Yes! good job 👍